src/components/ProductInfo/ProductInfo.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { getProductById } from '../../api/products';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import {
+  setError,
+  setIsLoading,
+  setProduct,
+} from '../../features/productSlice';
+import { Loader } from '../Loader';
+import styles from './ProductInfo.module.scss';
+
+export const ProductInfo: React.FC = () => {
+  const { productId = '' } = useParams();
+  const dispatch = useAppDispatch();
+  const { product, isLoading, error } = useAppSelector(
+    (state) => state.product
+  );
+
+  const propertiesData = [
+    {
+      name: 'Weight (imperial)',
+      value: product?.weight.imperial,
+    },
+    {
+      name: 'Weight (metric)',
+      value: product?.weight.metric,
+    },
+    {
+      name: 'Height (imperial)',
+      value: product?.height.imperial,
+    },
+    {
+      name: 'Height (metric)',
+      value: product?.height.metric,
+    },
+    {
+      name: 'Bred for',
+      value: product?.bred_for,
+    },
+    {
+      name: 'Breed group',
+      value: product?.breed_group,
+    },
+    {
+      name: 'Life span',
+      value: product?.life_span,
+    },
+    {
+      name: 'Temperament',
+      value: product?.temperament,
+    },
+    {
+      name: 'Origin',
+      value: product?.origin,
+    },
+  ];
+
+  const getProductFromServer = async (id: string) => {
+    try {
+      dispatch(setIsLoading(true));
+      dispatch(setError(''));
+
+      const productFromServer = await getProductById(Number(id));
+
+      if (productFromServer) {
+        dispatch(setProduct(productFromServer));
+      } else {
+        dispatch(setError('Product has not been found'));
+      }
+    } catch (error: any) {
+      dispatch(setError('Product has not been loaded. Try again later.'));
+    } finally {
+      dispatch(setIsLoading(false));
+    }
+  };
+
+  useEffect(() => {
+    getProductFromServer(productId);
+
+    window.addEventListener(
```

Comment:
You should always unsubscribe from the event Listener when component unmounts

Reply:
BTW, what this addEventListener used for?

Why we need to reload the page?

Reply:
@IgorNovikov10 because without this code, if you change the product id in the address bar in the browser - nothing will happen. but with this code - you will move to this product page with this id.

Reply:
Checked your code, why the '#" symbol appears in the address bar when navigating to the products page? It shoule not be there

If it's not presented in the url you won't need this hashchange event listener anymore

Reply:
@IgorNovikov10  '#' symbol appears because i use HashRouter.

Now on my last commit I change it to BrowserRouter and I have a problem - when I move to page "Products", for example, and press F5 button - I see 404 page. 
It happens because GitHub Pages - is a static files server. 
To prevent it - I found a decision with HashRouter, but you say that "#" symbol should not be in URL.

What to do in this case?